src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -167,12 +169,36 @@ public String getServiceClientConstructorName(Interface service) {
   /**
    * Returns the local name used in Go files for the package of the proto.
    */
-  private static String localPackageName(ProtoElement proto) {
+  private String localPackageName(ProtoElement proto) {
     String goPackage = proto.getFile().getProto().getOptions().getGoPackage();
-    if (goPackage.startsWith(CORE_PROTO_BASE)) {
-      return goPackage.substring(CORE_PROTO_BASE.length()).replace("/", "_");
+    if (!goPackage.startsWith(CORE_PROTO_BASE)) {
+      return proto.getFile().getProto().getPackage().replace(".", "_");
     }
-    return proto.getFile().getProto().getPackage().replace(".", "_");
+    // Inside CORE_PROTO_BASE, we are referencing code generated by curated protos.
+    // Import path of these protos might be versioned
+    //   google.golang.org/genproto/googleapis/example/library/v1
+    // or not
+    //   google.golang.org/genproto/googleapis/api/monitoredres
+    // We heuristically get the import name by looking for the right-most element,
+    // that is not a version number.
+    List<String> parts = Arrays.asList(goPackage.split("/"));
+    Collections.reverse(parts);
+    String name = parts.get(0);
```

Comment:
This section will end up with a weird result when full name consists of all v[0-9]+ patterns (like "v1/v2/b3"). Or if it's "golang.org/v1/v2", `name` becomes `golang.org` which will be invalid.  Such things would never happen in the realistic situation though.

Considering that, however, I think this line would be unnecessary; simply saying `String name = null;` is better, so it can fail with errors in case we encounter such erroneous situation.